### ISSUE TITLE
Digger apply fails to download artifact (plan file) from GitHub

### ISSUE THREAD

**Issue by damjanlepcheski** [diggerhq/digger#1068]
In our setup we specify 'github' as destination for the tfplan files. So when I run `digger plan`, the plan file is successfully attached to the PR with 'plan-$PRNUMBER.zip' name format. All good so far. 

However, when running `digger apply`, another `plan` is executed as part of the `digger run`, and this produced only local tfplan file, which is **not** uploaded to GitHub. When the `apply` step finally happens, it fails with the following errors:

```
Failed to run digger apply command. error retrieving plan: error downloading plan: failed to download artifact, status code: 400
Failed to run commands. error while running command: Failed to run digger apply command. <nil>
Error: Process completed with exit code 8.
```
When I remove the  `upload-plan-destination: 'github'` attribute from the workflow, `digger apply` is successful, likely because it is using the locally generated tfplan. 

### Additional helpful information 
- In the version I previously used (v0.2.0), `digger apply` successfully fetched the tfplan file from GitHub.
- Here's the workflow file I use:
```yaml
name: Digger
run-name: Digger on a Fargate hosted Github Actions Runner
on:
  pull_request:
    branches: [ "main" ]
    types: [ closed, opened, synchronize, reopened ]
  issue_comment:
    types: [created]
    if: contains(github.event.comment.body, 'digger')
  workflow_dispatch:

jobs:
  build:
    if: contains(github.event.comment.body, 'digger')
    runs-on:
      labels: default
    permissions:
      contents: write      # required to merge PRs
      id-token: write      # required for workload-identity-federation
      pull-requests: write # required to post PR comments
      statuses: write      # required to validate combined PR status
      actions: write       # required for artifact (plan) upload

    steps:
      - name: install-aws-cli
        uses: unfor19/install-aws-cli-action@v1
      - name: digger run
        uses: diggerhq/digger@v0.3.14
        with:
          setup-aws: false
          setup-terraform: true
          upload-plan-destination: 'github'
          no-backend: true
        env:
          GITHUB_CONTEXT: ${{ toJson(github) }}
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          GITHUB_WORKSPACE: ${{ github.workspace }}
```

I'm happy to provide more info if necessary.

**Comment by ZIJ** [comment]
Thanks for reporting - this is super helpful! We are looking into it

**Comment by ZIJ** [comment]
Some context from trying to fix it (also [asked](https://www.reddit.com/r/devops/comments/19d1wq8/github_actions_error_downloading_artifact/) on Reddit):

The [plan storage](https://github.com/diggerhq/digger/blob/58714b285cb0606597682157ac523e8ea65e47fc/cli/pkg/storage/plan_storage.go#L175) logic is fairly straightforward:

1. `DownloadLatestFiles()` function uses the ListArtifacts() method of the Github API client to list all artifacts for a repo

2. `getLatestArtifactWithName()` function picks the latest one for the PR matching plans-{prNumber} pattern

3. `DownloadArtifact()` method of the Github client is used to get the download URL of the artifact (NOT download the file). It returns 302 redirect ([docs](https://docs.github.com/en/rest/actions/artifacts?apiVersion=2022-11-28#download-an-artifact))

4. `downloadArtifactIntoFile()` function uses that URL to actually download the artifact into plans-{prNumber}.zip file

But for some reason, step 4 fails with a 400 error:

```
<?xml version="1.0" encoding="utf-8"?><Error><Code>InvalidAuthenticationInfo</Code><Message>Authentication information is not given in the correct format. Check the value of Authorization header.
RequestId:27a833fd-001e-00df-1052-4dd7c7000000
Time:2024-01-22T16:43:14.6875514Z</Message></Error>
```

The url expires in 1 minute so it’s supposed to be public. But just to be sure, I tried setting `Authorization: token <GITHUB_TOKEN>` in the request anyways; that didn’t help, still the same error (UPD neither did `bearer` format)

The URL in the redirect looks like this, presumably using Azure Blob Storage:

```
https://productionresultssa9.blob.core.windows.net/actions-results/<guid>/workflow-job-run-<guid>/artifacts/plans-11.zip?se=2024-01-22T17%3A18%3A40Z&sig=<signature>%3D&sp=r&spr=https&sr=b&st=2024-01-22T17%3A08%3A40Z&sv=2021-12-02
```

A similar error from blob storage like in this [question](https://stackoverflow.com/questions/22015813/authentication-not-in-correct-format-when-setting-azure-blob-service-propertie) on SO suggest using auth via SharedKey, but then it doesn’t really make sense in the context of Github Actions; it seems unlikely that GitHub API would return a URL that requires Azure-specific auth.

**Comment by ZIJ** [comment]
Should be fixed in #1081
